integrations/reporting_tools/prettytable_object.py

```diff
@@ -0,0 +1,38 @@
+"""
+A module to house prettytable custom objects
+This tail made objects are used by pytest to print table output of failure summary
+"""
+from prettytable.colortable import ColorTable, Theme, Themes
+
+#pylint: disable=too-few-public-methods
+class PrettyTableTheme(Themes):
+    "A custom color theme object"
+    Failure = Theme(default_color="31",
+                    vertical_color="31",
+                    horizontal_color="31",
+                    junction_color="31")
+
+class FailureSummaryTable():
+    "Failure Summary Table to be printed in the pytest result summary"
+    def __init__(self) -> None:
+        """
+        Initializer
+        """
+        # Create a pretty color table in red to mark failures
+        self.table = ColorTable(theme=PrettyTableTheme.Failure)
+        self.table.field_names = ["Tests Failed"]
+        self.table.align = "l" # <- Align the content of the table left
+
+    def print_table(self, testsummary: dict) -> None:
```

Comment:
> Would be good to add a check to ensure that testsummary is a dict, else raise ValueError.

This may not be needed. I tried to raise a `TypeError` but looked like an overkill.